include/ginkgo/core/base/mpi.hpp

```diff
@@ -567,7 +567,7 @@ class communicator {
      */
     bool is_identical(const communicator& rhs) const
     {
-        if (get() == nullptr || rhs.get() == nullptr) {
+        if (!get() || !rhs.get()) {
```

Comment:
TBH, maybe the first check is not necessary at all? I think it was in there to check if the `comm_` is a nullptr, but I think this can't be the case anymore.